track.py

```diff
@@ -266,6 +293,96 @@ def importSpeedLimitTuples(self, tuples, unit='km/h'):
 
 
+    def importCurvatureTuples(self, tuples, unitRadiusStart='m', unitRadiusEnd='m', clothoidSamplingInterval=None):
+
+        if not self.lengthOk():
+
+            raise ValueError("Cannot import curvature without a valid track length!")
+        
+        if not unitRadiusStart == unitRadiusEnd:
+
+            raise ValueError("Radius at start and radius at end must have same unit!")
+        
+        unit = unitRadiusStart
+
+        if unit not in {'m', 'km'}:
+
+            raise ValueError("Specified curvature radius unit not supported!")
+        
+        # if radius is 'infinity', the casting to float produces the float inf
+        tuples = [(p, convertUnit(float(radiusStart), unit), convertUnit(float(radiusEnd), unit)) for p, radiusStart, radiusEnd in tuples]
+
+        tuples = self.sampleClothoid(tuples, clothoidSamplingInterval)
+
+        self.curvatures = importTuples(tuples, 'Position [m]', ['Curvature [1/m]'])
+
+        checkDataFrame(self.curvatures, self.length)
+
+
+    def sampleClothoid(self, tuples, ds=None):
+        """
+        Approximates clothoid transition curves with piecewise-constant functions.
+
+        Given an interval [s_i, s_i + ds] the approximation of K(s) on the interval
+        is K_avg(s) = (K(s_i) + K(s_i + ds))/2. For the last interval [s_i, s_f] the 
+        approximation of K(s) is K_avg(s) = (K(s_i) + K(s_f))/2 where K(s_f) is the 
+        curvature at the end of the section. When ds is not specified, K(s) is 
+        approximated as K_approx(s) = (K(s_0) + K(s_f))/2.
+
+        - param tuples: a list of triples of form (p, Rstart, Rend) where p is the
+        coordinate at the start of the track section; Rstart is the radius at the 
+        start of the section and Rend the radius at the end.
+        - param ds: the step size used to approximate the clothoid. Note that in
+        general we can not guarantee all intervals to have size ds. Hence, in
+        general, the last interval has lenght L such that: ds <= L < 2*ds.
+        - return: a list of pairs (p, K) where K is the approximation 
+        of the clothoid curvature in the track section starting at position p.
+        """
+
+        result = []
+
+        epsilon = sys.float_info.epsilon
+
+        for trackIndex, trackSection in enumerate(tuples):
+
+            sectionStart = trackSection[0]
+            curvatureStart = 1/trackSection[1]
+            curvatureEnd = 1/trackSection[2]
+
+            if (curvatureStart - curvatureEnd) <= epsilon:
+
+                result.append((sectionStart, curvatureStart))
+
+            else:
+
+                sectionEnd = tuples[trackIndex+1][0] if trackIndex < len(tuples)-1 else self.length
+
+                if ds == None or int((sectionEnd-sectionStart)/ds) == 0:
+
+                    result.append((sectionStart, (curvatureStart + curvatureEnd)/2))
+
+                else:
+
+                    nIntervals = int((sectionEnd-sectionStart)/ds)
```

Comment:
Do we have a unit test that this is doing the right thing?

Reply:
I added the test  with the latest commit.I will close this thread but please have a look at the unit test.